### ISSUE TITLE
IntelliJ frontend unresponsive due to cloud code plugin

### ISSUE THREAD

**Issue by galderz** [GoogleCloudPlatform/cloud-code-intellij#2688]
IntelliJ fails to start due to Google Cloud Run plugin:

```
"AWT-EventQueue-0" #25 prio=6 os_prio=31 cpu=106126.57ms elapsed=236.93s tid=0x00007fed78953000 nid=0x13603 runnable  [0x000070000c23a000]
   java.lang.Thread.State: RUNNABLE
	at org.yaml.snakeyaml.scanner.ScannerImpl.scanPlain(ScannerImpl.java:2010)
	at org.yaml.snakeyaml.scanner.ScannerImpl.fetchPlain(ScannerImpl.java:1046)
	at org.yaml.snakeyaml.scanner.ScannerImpl.fetchMoreTokens(ScannerImpl.java:401)
	at org.yaml.snakeyaml.scanner.ScannerImpl.checkToken(ScannerImpl.java:226)
	at org.yaml.snakeyaml.parser.ParserImpl$ParseBlockMappingValue.produce(ParserImpl.java:585)
	at org.yaml.snakeyaml.parser.ParserImpl.peekEvent(ParserImpl.java:157)
	at org.yaml.snakeyaml.parser.ParserImpl.checkEvent(ParserImpl.java:147)
	at org.yaml.snakeyaml.composer.Composer.composeNode(Composer.java:133)
	at org.yaml.snakeyaml.composer.Composer.composeValueNode(Composer.java:246)
	at org.yaml.snakeyaml.composer.Composer.composeMappingChildren(Composer.java:237)
	at org.yaml.snakeyaml.composer.Composer.composeMappingNode(Composer.java:225)
	at org.yaml.snakeyaml.composer.Composer.composeNode(Composer.java:154)
	at org.yaml.snakeyaml.composer.Composer.composeSequenceNode(Composer.java:199)
	at org.yaml.snakeyaml.composer.Composer.composeNode(Composer.java:152)
	at org.yaml.snakeyaml.composer.Composer.composeValueNode(Composer.java:246)
	at org.yaml.snakeyaml.composer.Composer.composeMappingChildren(Composer.java:237)
	at org.yaml.snakeyaml.composer.Composer.composeMappingNode(Composer.java:225)
	at org.yaml.snakeyaml.composer.Composer.composeNode(Composer.java:154)
	at org.yaml.snakeyaml.composer.Composer.composeValueNode(Composer.java:246)
	at org.yaml.snakeyaml.composer.Composer.composeMappingChildren(Composer.java:237)
	at org.yaml.snakeyaml.composer.Composer.composeMappingNode(Composer.java:225)
	at org.yaml.snakeyaml.composer.Composer.composeNode(Composer.java:154)
	at org.yaml.snakeyaml.composer.Composer.composeDocument(Composer.java:122)
	at org.yaml.snakeyaml.composer.Composer.getSingleNode(Composer.java:105)
	at org.yaml.snakeyaml.constructor.BaseConstructor.getSingleData(BaseConstructor.java:122)
	at org.yaml.snakeyaml.Yaml.loadFromReader(Yaml.java:505)
	at org.yaml.snakeyaml.Yaml.load(Yaml.java:436)
	at com.google.cloud.tools.intellij.kubernetes.skaffold.SkaffoldYamlConfiguration$Companion.loadFromVirtualFileOrProject(SkaffoldYamlConfiguration.kt:55)
	at com.google.cloud.tools.intellij.kubernetes.skaffold.SkaffoldYamlConfiguration.<init>(SkaffoldYamlConfiguration.kt:64)
	at com.google.cloud.tools.intellij.kubernetes.skaffold.SkaffoldYamlConfiguration.<init>(SkaffoldYamlConfiguration.kt:63)
	at com.google.cloud.tools.intellij.kubernetes.skaffold.SkaffoldFileService.isSkaffoldFile(SkaffoldFileService.kt:51)
	at com.google.cloud.tools.intellij.kubernetes.skaffold.SkaffoldFileService$findSkaffoldFiles$1.compute(SkaffoldFileService.kt:103)
	at com.google.cloud.tools.intellij.kubernetes.skaffold.SkaffoldFileService$findSkaffoldFiles$1.compute(SkaffoldFileService.kt:37)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:807)
	at com.google.cloud.tools.intellij.kubernetes.skaffold.SkaffoldFileService.findSkaffoldFiles(SkaffoldFileService.kt:101)
	at com.google.cloud.tools.intellij.kubernetes.skaffold.SkaffoldConfigurationDetector$Companion.isSkaffoldConfigurationNeeded(SkaffoldConfigurationDetector.kt:80)
	at com.google.cloud.tools.intellij.kubernetes.skaffold.SkaffoldConfigurationDetector.showPromptForSkaffoldConfigurations(SkaffoldConfigurationDetector.kt:144)
	at com.google.cloud.tools.intellij.kubernetes.skaffold.SkaffoldConfigurationDetector.access$showPromptForSkaffoldConfigurations(SkaffoldConfigurationDetector.kt:54)
	at com.google.cloud.tools.intellij.kubernetes.skaffold.SkaffoldConfigurationDetector$projectOpened$1.run(SkaffoldConfigurationDetector.kt:95)
	at com.intellij.openapi.project.DumbServiceImpl.lambda$runWhenSmart$0(DumbServiceImpl.java:247)
```

Workaround: kill process and delete `Library/ApplicationSupport/IntelliJIdea2019.3/google-cloud-code` folder.

**Comment by etanshaul** [comment]
@galderz thanks for reporting. Do you know what version of cloud code you were using? This issue should be fixed in the latest version.

**Comment by galderz** [comment]
I don't know unfortunately. I had to remove the folder manually and when i tried to reinstall it again from IntelliJ it happened again, so I've given up for the time being... You can't be blocking the IDE! 🤦‍♂️

**Comment by etanshaul** [comment]
@galderz got it. I understand the frustration. It's hard to say with certainty without more details (e.g. knowing which version of the plugin gave you this freeze), but we think we fixed this (tracking here: https://github.com/GoogleCloudPlatform/cloud-code-intellij/issues/2674) in release 19.12.1.

I'll close this for now since we suspect it's fixed, but if you decide to give this another shot please feel free to reopen and we'll help out.

**Comment by galderz** [comment]
The issue still happens

**Comment by galderz** [comment]
https://gist.github.com/galderz/420f2749f1faf7fe3cf3f4acdad5d25b

**Comment by galderz** [comment]
With version 19.12.1

**Comment by etanshaul** [comment]
Reopened.

@galderz do you have a `skaffold.yaml` file in your project? What about any other yaml config files (e.g. for Kubernetes) that are very large? I'm not sure that this is the culprit, but this might help us narrow it down.

**Comment by etanshaul** [comment]
@galderz update:

After digging deeper we realized that we do have a fix implemented, but that I didn't make it on the `19.12.1` release. We are prioritizing a new release which should hopefully go out in the next day or two. Thanks again for bringing this to our attention.

**Comment by etanshaul** [comment]
This should be fixed now by release `20.1.1`: https://github.com/GoogleCloudPlatform/cloud-code-intellij/releases

Closing issue for now, but feel free to reopen as needed. thanks.
